ZoomEye is a cyberspace search engine.

nixawk opened the pull request:
<img width="925" alt="screen shot 2016-11-11 at 21 13 55" src="https://cloud.githubusercontent.com/assets/7352479/20235297/cea9deb0-a853-11e6-9a5d-12e1a6379cca.png">

<img width="594" alt="screen shot 2016-11-11 at 21 16 19" src="https://cloud.githubusercontent.com/assets/7352479/20235311/2ad8d920-a854-11e6-9754-6662c265275e.png">


<img width="1041" alt="screen shot 2016-11-11 at 21 15 20" src="https://cloud.githubusercontent.com/assets/7352479/20235302/fee9ec5a-a853-11e6-8231-a61f9f31a904.png">

techgaun commented:
good to go if you resolve the conflict @nixawk . It was removed on https://github.com/enaqx/awesome-pentest/commit/b6280c86b3f70a5e66ba8e7116a4036512e5e7fa as it went down at that time.

nixawk commented:
@techgaun Zoomeye is behind **CDN Server**. Browsers can support the javascript code, and redirect to the web. 

```
$ curl -v https://www.zoomeye.org/
*   Trying 111.202.98.126...
* TCP_NODELAY set
* Connected to www.zoomeye.org (111.202.98.126) port 443 (#0)
* TLS 1.2 connection using TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
* Server certificate: www.zoomeye.org
* Server certificate: COMODO RSA Domain Validation Secure Server CA
* Server certificate: COMODO RSA Certification Authority
> GET / HTTP/1.1
> Host: www.zoomeye.org
> User-Agent: curl/7.51.0
> Accept: */*
>
< HTTP/1.1 521
< Server: nginx
< Date: Tue, 27 Dec 2016 02:28:43 GMT
< Transfer-Encoding: chunked
< Connection: keep-alive
< Set-Cookie: __jsluid=f3d630e56773801a455cab8c5b9ba3e3; max-age=31536000; path=/; HttpOnly
<
<script>var x="2@16@addEventListener@dc@1500@function@charAt@catch@@838@0@@return@cd@pKGk@cookie@setTimeout@orders@4@43@attachEvent@join@window@03@for@B@document@GMT@var@@Array@length@Dec@27@7@Z@Tue@i@try@l@false@else@_phantom@replace@1482805723@F@href@28@__jsl_clearance@DOMContentLoaded@Path@Q@D@__phantomas@while@if@challenge@location@onreadystatechange@e@mqXST@6@Expires@captcha".replace(/@*$/,"").split("@"),y="t 17=6(){1m(n.1a||n.1l){};t e,4='1g=1c.a|b|';t i=[((+!~~[])+[[]][~~[]])+[1+([-~-~~~[]]+~~{}>>-~-~~~[])],[1t],(-~[12]+[]),((+!~~[])+[[]][~~[]])+(([-~-~~~[]]+~~{}>>-~-~~~[])+[]+[]),((+!~~[])+[[]][~~[]]),(1+[]+[]),((+!~~[])+[[]][~~[]])+(1+[]+[]),([(-~-~~~[]^-~[])]*((-~-~~~[]^-~[]))+[[], !+{}][(+[])]),(-~~~[]-~-~~~[]+[]+[[]][~~{}]),((+[])+[]+[]),((+!~~[])+[[]][~~[]])+(-~~~[]-~-~~~[]+[]+[[]][~~{}]),[1+([-~-~~~[]]+~~{}>>-~-~~~[])],((+!~~[])+[[]][~~[]])+(j+[[], []][-~[]]),(j+[[], []][-~[]]),(([-~-~~~[]]+~~{}>>-~-~~~[])+[]+[]),((+!~~[])+[[]][~~[]])+((+!~~[])+[[]][~~[]]),((+!~~[])+[[]][~~[]])+([(-~-~~~[]^-~[])]*((-~-~~~[]^-~[]))+[[], !+{}][(+[])]),((+!~~[])+[[]][~~[]])+[1t],((+!~~[])+[[]][~~[]])+((+[])+[]+[]),((+!~~[])+[[]][~~[]])+(-~[12]+[])];e=v(i.w);p(t 15=b;15<i.w;15++){e[i[15]]=['1j%',(-~!{}/~~!{}+[]+[]).7((-~{}+[1+1]>>-~{})),([]%(+[])+[]+[[]][~~{}]).7(-~[])+([(-~-~~~[]^-~[])]*((-~-~~~[]^-~[]))+[[], !+{}][(+[])])+[[-~{}-~{}]/~~{}+[]+[[]][~~{}]][b].7((+!~~[])+(-~{}-~{})*[-~{}-~{}]),'13','%',(1+[]+[]),(1+[]+[]),'f','1d',((+[])+[]+[])+((+!~~[])+[[]][~~[]])+(-~[12]+[]),'q','1s',([-~{}-~{}]/~~{}+[[]][~~[]]).7(~~!{}),(j+[[], []][-~[]]),'1k','%','1k',(-~~~[]-~-~~~[]+[]+[[]][~~{}]),([-~{}-~{}]/~~{}+[[]][~~[]]).7(~~!{})+[[][+[~~{}, ~~{}]]+[[], []][-~[]]][b].7(-~{})+[1+([-~-~~~[]]+~~{}>>-~-~~~[])],(-~~~[]-~-~~~[]+[]+[[]][~~{}])][15]};e=e.m('');4+=e;h('1p.1e=1p.1e.1b(/[\\?|&]1v-1o/,\\'\\')',5);r.g=(4+';1u=14, 11-10-2 o:1f:k s;1i=/;');};1n((6(){16{d !!n.3;}8(1r){d 18;}})()){r.3('1h',17,18);}19{r.l('1q',17);}",z=0,f=function(x,y){var a=0,b=0,c=0;x=x.split("");y=y||99;while((a=x.shift())&&(b=a.charCodeAt(0)-77.5))c=(Math.abs(b)<13?(b+48.5):parseInt(a,36))+y*c;return c},g=y.match(/\b\w+\b/g).sort(function(x,y){return f(x)-f(y)}).pop();while(f(g,++z)-x.length){};eval(y.replace(/\b\w+\b/g, function(y){return x[f(y,z)-1]}));</script>
* Curl_http_done: called premature == 0
* Connection #0 to host www.zoomeye.org left intact
```

The issue may not be solved.

- [**Zoomeye Python Library**](https://github.com/zoomeye/SDK) is available.

techgaun commented:
seems that one way to do this is to probably allow http status code _521_. but then there could be other services behind cloudflare and it seems that 521 is used by CF to indicate backend server unavailable.. not sure how we can approach this best so the awesome_bot check does not miss bad ones behind CF.

techgaun commented:
Adding zoomeye as part of another commit
